crates/solver/src/driver.rs

```diff
@@ -381,8 +382,10 @@ impl Driver {
                             .unwrap_or(f64::NAN),
                         surplus: rated_settlement.surplus.to_f64().unwrap_or(f64::NAN),
                         fees: rated_settlement.solver_fees.to_f64().unwrap_or(f64::NAN),
-                        cost: rated_settlement.gas_estimate.to_f64_lossy()
-                            * rated_settlement.gas_price.to_f64().unwrap_or(f64::NAN),
+                        cost: match &rated_settlement.gas_cost {
+                            solver_settlements::GasCost::Protocol(gas_cost) => Cost::Protocol(gas_cost.to_f64().unwrap_or(f64::NAN)),
```

Comment:
This should probably not silently fail but instead cause a proper error

Reply:
Hm, makes sense although I just followed similar handling of other conversions.
Skipping failures is probably ok here since the data is used for debugging purposes and not important for the rest of the `single_function` execution.